Build failure on i586 plattform

Hi,

nice project, BTW. While packaging for openSUSE, I came across a build issue:

```
[   15s] Executing(%build): /bin/sh -e /var/tmp/rpm-tmp.RKb8e4
[   15s] + umask 022
[   15s] + cd /home/abuild/rpmbuild/BUILD
[   15s] + /usr/bin/rm -rf /home/abuild/rpmbuild/BUILDROOT/python-poppler-0.2.1-0.i386
[   15s] ++ dirname /home/abuild/rpmbuild/BUILDROOT/python-poppler-0.2.1-0.i386
[   15s] + /usr/bin/mkdir -p /home/abuild/rpmbuild/BUILDROOT
[   15s] + /usr/bin/mkdir /home/abuild/rpmbuild/BUILDROOT/python-poppler-0.2.1-0.i386
[   15s] + cd python-poppler-0.2.1
[   15s] ++ '[' -f _current_flavor ']'
[   15s] ++ true
[   15s] + python_flavor=
[   15s] + '[' -z '' ']'
[   15s] + python_flavor=tmp
[   15s] + '[' tmp '!=' python3 ']'
[   15s] + '[' -d build ']'
[   15s] + '[' -d _build.python3 ']'
[   15s] + echo python3
[   15s] + /usr/bin/python3 setup.py build '--executable=/usr/bin/python3 -s'
[   15s] running build
[   15s] running build_py
[   15s] creating build
[   15s] creating build/lib.linux-i686-3.8
[   15s] creating build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/__init__.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/_version.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/destination.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/document.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/embeddedfile.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/font.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/image.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/page.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/pagerenderer.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/pagetransition.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/rectangle.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/toc.py -> build/lib.linux-i686-3.8/poppler
[   15s] copying src/poppler/utilities.py -> build/lib.linux-i686-3.8/poppler
[   15s] creating build/lib.linux-i686-3.8/poppler/cpp
[   15s] copying src/poppler/cpp/__init__.py -> build/lib.linux-i686-3.8/poppler/cpp
[   15s] running egg_info
[   15s] writing src/python_poppler.egg-info/PKG-INFO
[   15s] writing dependency_links to src/python_poppler.egg-info/dependency_links.txt
[   15s] writing top-level names to src/python_poppler.egg-info/top_level.txt
[   15s] reading manifest file 'src/python_poppler.egg-info/SOURCES.txt'
[   15s] reading manifest template 'MANIFEST.in'
[   15s] writing manifest file 'src/python_poppler.egg-info/SOURCES.txt'
[   15s] running build_ext
[   15s] -- The C compiler identification is GNU 10.2.1
[   15s] -- The CXX compiler identification is GNU 10.2.1
[   15s] -- Detecting C compiler ABI info
[   15s] -- Detecting C compiler ABI info - done
[   15s] -- Check for working C compiler: /usr/bin/cc - skipped
[   15s] -- Detecting C compile features
[   15s] -- Detecting C compile features - done
[   15s] -- Detecting CXX compiler ABI info
[   16s] -- Detecting CXX compiler ABI info - done
[   16s] -- Check for working CXX compiler: /usr/bin/c++ - skipped
[   16s] -- Detecting CXX compile features
[   16s] -- Detecting CXX compile features - done
[   16s] -- Found PythonInterp: /usr/bin/python3 (found version "3.8.5") 
[   16s] -- Found PythonLibs: /usr/lib/libpython3.8.so
[   16s] -- Performing Test HAS_CPP14_FLAG
[   16s] -- Performing Test HAS_CPP14_FLAG - Success
[   16s] -- pybind11 v2.5.0
[   16s] -- Found PkgConfig: /usr/bin/pkg-config (found version "1.7.3") 
[   16s] -- Checking for module 'poppler-cpp>=0.62.0'
[   16s] --   Found poppler-cpp, version 0.90.0
[   16s] -- Configuring done
[   16s] -- Generating done
[   16s] -- Build files have been written to: /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/build/temp.linux-i686-3.8
[   16s] Scanning dependencies of target global_
[   16s] Scanning dependencies of target version
[   16s] [  8%] Building CXX object CMakeFiles/version.dir/src/cpp/version.cpp.o
[   16s] [  8%] Building CXX object CMakeFiles/global_.dir/src/cpp/global.cpp.o
[   19s] [ 12%] Linking CXX shared module ../lib.linux-i686-3.8/poppler/cpp/version.cpython-38-i386-linux-gnu.so
[   19s] [ 12%] Built target version
[   19s] Scanning dependencies of target image
[   19s] [ 16%] Building CXX object CMakeFiles/image.dir/src/cpp/image.cpp.o
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/src/cpp/image.cpp: In function ‘pybind11::buffer_info poppler::image_buffer_info(poppler::image&)’:
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/src/cpp/image.cpp:54:5: error: no matching function for call to ‘pybind11::buffer_info::buffer_info(void*, long int, std::string, long int, <brace-encl’
[   19s]    54 |     );
[   19s]       |     ^
[   19s] In file included from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/pytypes.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/cast.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/attr.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/pybind11.h:44,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/src/cpp/image.cpp:20:
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:89:5: note: candidate: ‘pybind11::buffer_info::buffer_info(pybind11::buffer_info::private_ctr_tag, void*, pybin’
[   19s]    89 |     buffer_info(private_ctr_tag, void *ptr, ssize_t itemsize, const std::string &format, ssize_t ndim,
[   19s]       |     ^~~~~~~~~~~
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:89:5: note:   candidate expects 8 arguments, 6 provided
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:64:5: note: candidate: ‘pybind11::buffer_info::buffer_info(pybind11::buffer_info&&)’
[   19s]    64 |     buffer_info(buffer_info &&other) {
[   19s]       |     ^~~~~~~~~~~
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:64:5: note:   candidate expects 1 argument, 6 provided
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:54:14: note: candidate: ‘pybind11::buffer_info::buffer_info(Py_buffer*, bool)’
[   19s]    54 |     explicit buffer_info(Py_buffer *view, bool ownview = true)
[   19s]       |              ^~~~~~~~~~~
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:54:14: note:   candidate expects 2 arguments, 6 provided
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:51:5: note: candidate: ‘template<class T> pybind11::buffer_info::buffer_info(const T*, pybind11::ssize_t, bool)’
[   19s]    51 |     buffer_info(const T *ptr, ssize_t size, bool readonly=true)
[   19s]       |     ^~~~~~~~~~~
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:51:5: note:   template argument deduction/substitution failed:
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/src/cpp/image.cpp:54:5: note:   candidate expects 3 arguments, 6 provided
[   19s]    54 |     );
[   19s]       |     ^
[   19s] In file included from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/pytypes.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/cast.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/attr.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/pybind11.h:44,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/src/cpp/image.cpp:20:
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:47:5: note: candidate: ‘template<class T> pybind11::buffer_info::buffer_info(T*, pybind11::ssize_t, bool)’
[   19s]    47 |     buffer_info(T *ptr, ssize_t size, bool readonly=false)
[   19s]       |     ^~~~~~~~~~~
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:47:5: note:   template argument deduction/substitution failed:
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/src/cpp/image.cpp:54:5: note:   candidate expects 3 arguments, 6 provided
[   19s]    54 |     );
[   19s]       |     ^
[   19s] In file included from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/pytypes.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/cast.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/attr.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/pybind11.h:44,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/src/cpp/image.cpp:20:
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:43:5: note: candidate: ‘pybind11::buffer_info::buffer_info(void*, pybind11::ssize_t, const string&, pybind11::s’
[   19s]    43 |     buffer_info(void *ptr, ssize_t itemsize, const std::string &format, ssize_t size, bool readonly=false)
[   19s]       |     ^~~~~~~~~~~
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:43:5: note:   candidate expects 5 arguments, 6 provided
[   19s] In file included from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/pytypes.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/cast.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/attr.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/pybind11.h:44,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/src/cpp/image.cpp:20:
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:40:5: note: candidate: ‘template<class T> pybind11::buffer_info::buffer_info(T*, pybind11::detail::any_containe’
[   19s]    40 |     buffer_info(T *ptr, detail::any_container<ssize_t> shape_in, detail::any_container<ssize_t> strides_in, bool readonly=false)
[   19s]       |     ^~~~~~~~~~~
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:40:5: note:   template argument deduction/substitution failed:
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/src/cpp/image.cpp:54:5: note:   candidate expects 4 arguments, 6 provided
[   19s]    54 |     );
[   19s]       |     ^
[   19s] In file included from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/pytypes.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/cast.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/attr.h:13,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/pybind11.h:44,
[   19s]                  from /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/src/cpp/image.cpp:20:
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:29:5: note: candidate: ‘pybind11::buffer_info::buffer_info(void*, pybind11::ssize_t, const string&, pybind11::s’
[   19s]    29 |     buffer_info(void *ptr, ssize_t itemsize, const std::string &format, ssize_t ndim,
[   19s]       |     ^~~~~~~~~~~
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:30:89: note:   no known conversion for argument 6 from ‘<brace-enclosed initializer list>’ to ‘pybind11::detail’
[   19s]    30 |                 detail::any_container<ssize_t> shape_in, detail::any_container<ssize_t> strides_in, bool readonly=false)
[   19s]       |                                                          ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:27:5: note: candidate: ‘pybind11::buffer_info::buffer_info()’
[   19s]    27 |     buffer_info() { }
[   19s]       |     ^~~~~~~~~~~
[   19s] /home/abuild/rpmbuild/BUILD/python-poppler-0.2.1/pybind11/include/pybind11/buffer_info.h:27:5: note:   candidate expects 0 arguments, 6 provided
[   20s] gmake[2]: *** [CMakeFiles/image.dir/build.make:82: CMakeFiles/image.dir/src/cpp/image.cpp.o] Error 1
[   20s] gmake[1]: *** [CMakeFiles/Makefile2:191: CMakeFiles/image.dir/all] Error 2
[   20s] gmake[1]: *** Waiting for unfinished jobs....
[   21s] [ 20%] Linking CXX shared module ../lib.linux-i686-3.8/poppler/cpp/global_.cpython-38-i386-linux-gnu.so
[   21s] [ 20%] Built target global_
[   21s] gmake: *** [Makefile:103: all] Error 2
[   21s] Traceback (most recent call last):
[   21s]   File "setup.py", line 76, in <module>
[   21s]     setup(
[   21s]   File "/usr/lib/python3.8/site-packages/setuptools/__init__.py", line 162, in setup
[   21s]     return distutils.core.setup(**attrs)
[   21s]   File "/usr/lib/python3.8/distutils/core.py", line 148, in setup
[   21s]     dist.run_commands()
[   21s]   File "/usr/lib/python3.8/distutils/dist.py", line 966, in run_commands
[   21s]     self.run_command(cmd)
[   21s]   File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
[   21s]     cmd_obj.run()
[   21s]   File "/usr/lib/python3.8/distutils/command/build.py", line 135, in run
[   21s]     self.run_command(cmd_name)
[   21s]   File "/usr/lib/python3.8/distutils/cmd.py", line 313, in run_command
[   21s]     self.distribution.run_command(command)
[   21s]   File "/usr/lib/python3.8/distutils/dist.py", line 985, in run_command
[   21s]     cmd_obj.run()
[   21s]   File "setup.py", line 39, in run
[   21s]     self.build_extension(ext)
[   21s]   File "setup.py", line 71, in build_extension
[   21s]     subprocess.check_call(
[   21s]   File "/usr/lib/python3.8/subprocess.py", line 364, in check_call
[   21s]     raise CalledProcessError(retcode, cmd)
[   21s] subprocess.CalledProcessError: Command '['cmake', '--build', '.', '--config', 'Release', '--', '-j2']' returned non-zero exit status 2.
[   21s] error: Bad exit status from /var/tmp/rpm-tmp.RKb8e4 (%build)
```

Obviously, the 32bit signature of `pybind11::buffer_info::buffer_info` differs somehow. 

Full build is available [here](https://build.opensuse.org/package/show/home:frispete:python/python-poppler).

Sorry, been in a hurry ATM, therefor no PR for now.

--- a/src/cpp/image.cpp
+++ b/src/cpp/image.cpp
@@ -50,7 +50,7 @@ py::buffer_info image_buffer_info(image &img)
         py::format_descriptor<char>::format(),
         3L,
         { static_cast<py::ssize_t>(img.height()), static_cast<py::ssize_t>(img.width()), bytes_per_color },
-        { static_cast<py::ssize_t>(img.bytes_per_row()), bytes_per_color, 1L }
+        { static_cast<py::ssize_t>(img.bytes_per_row()), bytes_per_color, static_cast<py::ssize_t>(1L) }
     );
 }
 

